test/unit/org/apache/cassandra/db/DirectoriesTest.java

```diff
@@ -136,7 +137,7 @@ private static void createTestFiles() throws IOException
 
     private static void createFakeSSTable(File dir, String cf, int gen, List<File> addTo) throws IOException
     {
-        Descriptor desc = new Descriptor(dir, KS, cf, gen, SSTableFormat.Type.BIG);
+        Descriptor desc = new Descriptor(dir, KS, cf, new SequenceBasedSSTableUniqueIdentifier(gen), SSTableFormat.Type.BIG);
```

Comment:
ULID is used in most cases, especially when we start Cassandra as there is no way to change the factory and we always use ULID. In some tests, which does not really test the identifier, but need to refer certain file names I left int based identifier because it is easy to track the generated file names. 

For some tests your idea looks valid though, perhaps `DirectoriesTest` is a good example